Jenkinsfile

```diff
@@ -0,0 +1,113 @@
+#!groovy
+
+// https://github.com/feedhenry/fh-pipeline-library
+@Library('fh-pipeline-library') _
+
+stage('Trust') {
+    enforceTrustedApproval('aerogear')
+}
+
+def operatorName = "unifiedpush-operator"
+def openshiftProjectName = "test-${operatorName}-${currentBuild.number}-${currentBuild.startTimeInMillis}"
+def operatorDockerImageName = "docker-registry.default.svc:5000/${openshiftProjectName}/${operatorName}-test:latest"
+def testFileName = 'go-test.sh'
+
+def testFileContent = """#!/bin/sh
+${operatorName}-test -test.parallel=1 -test.failfast -root=/ -kubeconfig=incluster -namespacedMan=namespaced.yaml -test.v
+"""
+
+def dockerfileContent = """
+FROM registry.access.redhat.com/ubi7-dev-preview/ubi-minimal:7.6
```

Comment:
Would it be better or worse to have this as a file in the repo, rather than writing it here for every run? My thinking is that it might make it more obvious that if we make changes to the main Dockerfile, that this should probably change too.

Alternatively, what about building the main image first, then use that as the `FROM` in the test image that just adds the test stuff? That way we could have this Jenkinsfile be the thing that builds the images that we push to quay.io: we'd be publishing images that we've run e2e tests against, and we'd have built them on infrastructure that we trust more than travis!

I'm not saying that any of this is stuff that we need to do here in this PR, just ideas for the future! :)

Reply:
This is actually a great idea and I didn't realise that we could [push the image built by OpenShift](https://blog.openshift.com/pushing-application-images-to-an-external-registry/) right to quay.io. I will create a new jira for this.